tests/integration/security/util/reachability/context.go

```diff
@@ -175,64 +107,92 @@ func (rc *Context) Run(testCases []TestCase) {
 			ctx.Logf("[%s] [%v] Finish waiting. Continue testing.", testName, time.Now())
 			ctx.Cleanup(func() {
 				if err := retry.UntilSuccess(func() error {
-					return rc.ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
+					return ctx.Config().DeleteYAML(c.Namespace.Name(), policyYAML)
 				}); err != nil {
 					log.Errorf("failed to delete configuration: %v", err)
 				}
 			})
-
-			for _, src := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Naked, rc.HeadlessNaked} {
-				for _, dest := range []echo.Instance{rc.A, rc.B, rc.Headless, rc.Multiversion, rc.Naked, rc.VM, rc.HeadlessNaked} {
-					copts := &callOptions
-					// If test case specified service call options, use that instead.
-					if c.CallOpts != nil {
-						copts = &c.CallOpts
+			for _, srcCluster := range ctx.Clusters() {
+				for _, dstCluster := range ctx.Clusters() {
```

Comment:
The `dstCluster` doesn't really apply here. The destination instances fall under the same service name per-cluster. Instead of looping over destination instances, just take the first instance make the call with `Count` > `2 * len(clusters)`. In the checker, make sure we load balance across all clusters. 


Grabbing first instance of a "destination set":

https://github.com/istio/istio/blob/379eb242402dfec83f937c66e9dcfb38b8c5fe9b/tests/integration/pilot/common/routing.go#L385-L399

Validating we loadbalanced across all clusters using:

https://github.com/istio/istio/blob/379eb242402dfec83f937c66e9dcfb38b8c5fe9b/pkg/test/echo/client/parsedresponse.go#L184

Example

```
res.CheckReachedAllClusters(destinationSet.Clusters())
```